### PR TITLE
restored fdroid build flavor

### PR DESCRIPTION
**Pull request by mnbogner**
restored fdroid build flavor in gradle files and copied greatfire assets to fdroid folder.  also updated version name/code to reflect actual version.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -165,6 +168,12 @@ android {
             versionName computeVersionName('greatfire')
             // signingConfig signingConfigs.prod
         }
+        fdroid {
+            applicationId 'org.greatfire.wikiunblocked'
```
Comment (eighthave) [inline app/build.gradle]: Since the f-droid.org version will be signed by the f-droid.org APK signing key, I recommend using a specific Application ID so there isn't update/signature conflicts:
```suggestion
            applicationId 'org.greatfire.wikiunblocked.fdroid'
```

File: app/build.gradle [hunk 2]
```diff
@@ -275,6 +285,7 @@ dependencies {
     alphaImplementation "com.google.firebase:firebase-messaging-ktx:$firebaseMessagingVersion"
     betaImplementation "com.google.firebase:firebase-messaging-ktx:$firebaseMessagingVersion"
     greatfireImplementation "com.google.firebase:firebase-messaging-ktx:$firebaseMessagingVersion"
+    fdroidImplementation "com.google.firebase:firebase-messaging-ktx:$firebaseMessagingVersion"
```
Comment (eighthave) [inline app/build.gradle]: I don't know if this ends up in the final APK, but `com.google.mlkit` and `com.google.firebase` are both proprietary libs, so can't be included in f-droid.org.

File: app/build.gradle [hunk 3]
```diff
@@ -94,6 +96,7 @@ android {
         alpha { java.srcDirs += 'src/extra/java' }
         beta { java.srcDirs += 'src/extra/java' }
         greatfire { java.srcDirs += 'src/extra/java' }
+        fdroid { java.srcDirs += 'src/extra/java' }
```
Comment (eighthave) [inline app/build.gradle]: I think it is possible to do something like this for the _res_ files, so that they don't need to be copied.  But copying seems fine too since its expedient.

File: app/src/fdroid/google-services.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+              "client_id": "296120793014-8l3acuf7qvvesvqngoc06j2nk2v0bhin.apps.googleusercontent.com",
+              "client_type": 3
+            }
+          ]
+        }
+      }
+    }
+  ],
+  "configuration_version": "1"
+}
```
Comment (eighthave) [inline app/src/fdroid/google-services.json:40]: I'm guessing that this file would be unneeded if the _firebase_ and _mlkit_ deps are removed from the _fdroid_ flavor.

File: app/build.gradle (hunk 2 above)
Comment (mnbogner) [reply app/build.gradle]: it looks like mlkit is being used to auto-detect languages when editing descriptions.

Comment (mnbogner) [reply app/build.gradle]: firebase messaging also seems to be used for some kind of notifications if users are logged in.

Comment (mnbogner) [reply app/build.gradle]: in the dependency list here i see both mlkit and firebase but only mlkit is flagged: https://gitlab.com/fdroid/fdroiddata/-/merge_requests/12056  in the "Scan APK" list i only see https://pushnotifications-73c5e.firebaseio.com though

File: app/build.gradle (hunk 1 above)
Comment (mnbogner) [reply app/build.gradle]: changing this required the secrets file to be rebuilt because it's based on the package name.  between this and the proprietary dependencies we may need to create a new branch for f-droid to build from.

File: app/build.gradle (hunk 2 above)
Comment (mnbogner) [reply app/build.gradle]: i commented out all of the code that relates to these libraries and it didn't seem to impact logins or notifications within the app.  i'm not really familiar with wikipedia user features so i'm not sure if i can test it more extensively.  as mentioned above, the other question would be whether to put these changes into a new branch for f-droid to build from.

File: app/src/fdroid/google-services.json (hunk 4 above)
Comment (mnbogner) [reply app/src/fdroid/google-services.json:40]: it also requires the removal of the "apply plugin: 'com.google.gms.google-services'" line from the apps build.gradle file

Comment (eighthave) [reply app/src/fdroid/google-services.json:40]: The wikipedia app is already on fdroid, so the greatfire fdroid should just be as close as possible to that.  I vaguely remember something about it having references to proprietary libs, but then they weren't included in the final APK.

File: app/build.gradle (hunk 1 above)
Comment (mnbogner) [reply app/build.gradle]: it seems like the best way to fix this is to remove the ".fdroid" suffix when i fetch the secrets in the code.  i couldn't find a way to reference a different set of native files for each build flavor.

File: app/src/fdroid/google-services.json (hunk 4 above)
Comment (mnbogner) [reply app/src/fdroid/google-services.json:40]: it looks as if the original wikipedia build.gradle doesn't assign fdroid an applicationIdSuffix or include a unique google-services.json file in the fdroid directory, so it just relies on the default file in the app folder: https://github.com/wikimedia/apps-android-wikipedia/blob/main/app/build.gradle#L132  that file doesn't include our package name though, so i'll need to keep this version of the file since i don't think i can remove the google-services plugin from the shared build.gradle file.

i'm hoping that by not including the mlkit and firebase dependencies in the fdroid build flavor, and by using the stubbed out version of the classes that required them, that the packages will be excluded from the final apk.

File: app/build.gradle (hunk 2 above)
Comment (mnbogner) [reply app/build.gradle]: it looks like the solution in the original wikipedia build.gradle was not to add these dependencies for the fdroid flavor and to include stubbed out versions of the classes that relied on them.  i'll do the same here.

File: app/build.gradle (hunk 3 above)
Comment (mnbogner) [reply app/build.gradle]: it looks like the src/extra/java directory is where the non-stubbed versions of the mlkit and firebase classes exist.  i assume by adding that path for every flavor but fdroid it includes the classes that require the proprietary dependencies in every non-fdroid flavor.

File: app/build.gradle [hunk 5]
```diff
@@ -21,9 +21,8 @@ def static getDate() {
     return formattedDate
 }
 
-// TODO - increment version? use parameter?
 def computeVersionName(label) {
-    return "2.7.${android.defaultConfig.versionCode}-${label}-${date}"
+    return "${android.defaultConfig.versionName}-${label}"
```
Comment (vitriolix) [inline app/build.gradle:25]: why the change to versioning scheme?

**Comment by vitriolix** [comment]
looks ok to me, didn't get a chance to run it though
